plux/runtime/filter.py

```diff
@@ -0,0 +1,83 @@
+import fnmatch
+import logging
+import typing as t
+
+from plux.core.entrypoint import spec_to_entry_point
+from plux.core.plugin import PluginSpec
+
+LOG = logging.getLogger(__name__)
+
+
+class PluginFilter(t.Protocol):
+    def __call__(self, spec: PluginSpec) -> bool:
+        """
+        Returns True if the plugin should be filtered (disabled), or False otherwise.
+
+        :param spec: the spec to check
+        :return: True if the plugin should be disabled
+        """
+        ...
+
+
+class PluginSpecMatcher:
+    namespace: str = None
+    name: str = None
+    value: str = None
+
+    def __init__(self, namespace: str = None, name: str = None, value: str = None):
+        self.namespace = namespace
+        self.name = name
+        self.value = value
+
+    def matches(self, spec: PluginSpec) -> bool:
+        if self.namespace:
+            if not fnmatch.fnmatch(spec.namespace, self.namespace):
+                return False
+
+        if self.name:
+            if not fnmatch.fnmatch(spec.name, self.name):
+                return False
+
+        if self.value:
+            ep = spec_to_entry_point(spec)
+            if not fnmatch.fnmatch(ep.value, self.value):
+                return False
+
+        return True
+
+
+class MatchingPluginFilter:
+    matchers: t.List[PluginSpecMatcher]
+
+    def __init__(self):
+        self.matchers = []
+
+    def add_pattern(self, namespace: str = None, name: str = None, value: str = None):
```

Comment:
nit: Given the type and structure of the arguments here, I think it might be clearer to only allow kwargs here and not any positional arguments :man_shrugging:  The motivation for this comes from the tests below where IMO it's not very clear if it's really intended to add a namespace pattern vs. a value pattern